File: scripts/loadOsduData.py
import os
import json
import re
from pathlib import Path

"""
Python script to load OSDU data from your actual data definitions folder
Run this to generate a data file for the React app
"""

OSDU_BASE_PATH = r"c:\Users\mzptto\OneDrive - amazon.com\EDI\OSDU Data Model\datadefinitions"

def load_osdu_entities():
    entities = []
    
    try:
        # Get all entity types from master-data
        master_data_path = Path(OSDU_BASE_PATH) / "E-R" / "_diagrams" / "master-data"
        
        if not master_data_path.exists():
            print(f"Path not found: {master_data_path}")
            return
            
        puml_files = [f for f in master_data_path.glob("*.puml") if ".ref." not in f.name]
        
        for puml_file in puml_files:
            match = re.match(r'^(\w+)\.(\d+\.\d+\.\d+)\.puml$', puml_file.name)
            if not match:
                continue
                
            entity_name, version = match.groups()
            
            try:
                # Load PlantUML content
                with open(puml_file, 'r', encoding='utf-8') as f:
                    puml_content = f.read()
                
                # Load schema
                schema_path = Path(OSDU_BASE_PATH) / "Generated" / "master-data" / f"{entity_name}.{version}.json"
                with open(schema_path, 'r', encoding='utf-8') as f:
                    schema = json.load(f)
                
                # Load example
                example_path = Path(OSDU_BASE_PATH) / "Examples" / "master-data" / f"{entity_name}.{version}.json"
                with open(example_path, 'r', encoding='utf-8') as f:
                    example = json.load(f)
                
                entities.append({
                    "id": f"{entity_name.lower()}-{version}",
                    "name": entity_name,
                    "type": "master-data",
                    "version": version,
                    "pumlContent": puml_content,
                    "schema": schema,
                    "example": example
                })
                
                print(f"Loaded: {entity_name} v{version}")
                
            except Exception as error:
                print(f"Failed to load {entity_name} v{version}: {error}")
        
        # Save to data file
        output_path = Path(__file__).parent.parent / "src" / "data" / "osduEntities.js"
        output_content = f"""// Auto-generated OSDU entities data
// Generated on {json.dumps(str(Path(__file__).stat().st_mtime))}

export const osduEntities = {json.dumps(entities, indent=2)};
"""
        
        with open(output_path, 'w', encoding='utf-8') as f:
            f.write(output_content)
            
        print(f"\nGenerated {len(entities)} entities to src/data/osduEntities.js")
        
    except Exception as error:
        print(f"Error loading OSDU data: {error}")

if __name__ == "__main__":
    load_osdu_entities()

File: scripts/loadOsduData.js
const fs = require('fs').promises;
const path = require('path');

/**
 * Script to load OSDU data from your actual data definitions folder
 * Run this to generate a data file for the React app
 */

const OSDU_BASE_PATH = 'C:\\Users\\mzptto\\OneDrive - amazon.com\\EDI\\OSDU Data Model\\datadefinitions';
const DIAGRAMS_PATH = path.join(OSDU_BASE_PATH, 'E-R', '_diagrams');

async function loadOsduEntities() {
  const entities = [];
  
  try {
    // Get all entity types from diagrams directory
    const entityTypes = await fs.readdir(DIAGRAMS_PATH);
    
    for (const entityType of entityTypes) {
      const entityTypePath = path.join(DIAGRAMS_PATH, entityType);
      const stat = await fs.stat(entityTypePath);
      
      if (!stat.isDirectory() || entityType === 'abstract' || entityType === 'content' || entityType === 'type' || entityType === 'FragmentRelations') continue;
      
      console.log(`Loading ${entityType} entities...`);
      
      try {
        const pumlFiles = await fs.readdir(entityTypePath);
        
        for (const pumlFile of pumlFiles) {
          if (!pumlFile.endsWith('.puml') || pumlFile.includes('.ref.')) continue;
          
          const match = pumlFile.match(/^([A-Za-z0-9\.]+)\.(\d+\.\d+\.\d+)\.puml$/);
          if (!match) continue;
          
          const [, entityName, version] = match;
          
          try {
            // Load PlantUML content
            const pumlPath = path.join(entityTypePath, pumlFile);
            const pumlContent = await fs.readFile(pumlPath, 'utf8');
            
            // Try to load schema
            let schema = null;
            try {
              const schemaPath = path.join(OSDU_BASE_PATH, 'Generated', entityType, `${entityName}.${version}.json`);
              const schemaContent = await fs.readFile(schemaPath, 'utf8');
              schema = JSON.parse(schemaContent);
            } catch (error) {
              // Schema not found, continue without it
            }
            
            // Try to load example
            let example = null;
            try {
              const examplePath = path.join(OSDU_BASE_PATH, 'Examples', entityType, `${entityName}.${version}.json`);
              const exampleContent = await fs.readFile(examplePath, 'utf8');
              example = JSON.parse(exampleContent);
            } catch (error) {
              // Example not found, continue without it
            }
            
            entities.push({
              id: `${entityName.toLowerCase()}-${version}`,
              name: entityName,
              type: entityType,
              version: version,
              pumlContent: pumlContent,
              schema: schema,
              example: example
            });
            
            console.log(`Loaded: ${entityName} v${version} (${entityType})`);
            
          } catch (error) {
            console.warn(`Failed to load ${entityName} v${version} (${entityType}):`, error.message);
          }
        }
      } catch (error) {
        console.warn(`Could not read ${entityType} directory:`, error.message);
      }
    }
    
    // Save to data file
    const outputPath = path.join(__dirname, '..', 'src', 'data', 'osduEntities.js');
    const outputContent = `// Auto-generated OSDU entities data
// Generated on ${new Date().toISOString()}

export const osduEntities = ${JSON.stringify(entities, null, 2)};
`;
    
    await fs.writeFile(outputPath, outputContent);
    console.log(`\\nGenerated ${entities.length} entities to src/data/osduEntities.js`);
    
  } catch (error) {
    console.error('Error loading OSDU data:', error);
  }
}

// Run if called directly
if (require.main === module) {
  loadOsduEntities();
}

module.exports = { loadOsduEntities };